### DISCUSSION TITLE
Questions about converting a LAMMPS .dump file to format usable by DeePMD-Kit

### DISCUSSION THREAD

**Discussion by agvent** [deepmodeling/deepmd-kit#4303] [Q&A] [answered]
Hello,

I have a LAMMPS .dump file containing atomic positions and forces, and I would like to convert the .dump file to a format usable by DeePMD-Kit using dpdata. I have a few questions:

1. According to the [LAMMPS documentation,](https://docs.lammps.org/dump.html) .dump files only contain per-atom data. I was unable to output my system wide data (energy and virial) to the .dump file. I instead have a separate file containing the energy and virial.  **Is there a way to add the system wide data into the .dump file? If not, how can I convert this file into the appropriate format for use with DeePMD-Kit?**
2. I've noticed that my .dump file does not always have the list of atomic positions and forces for each atom in the same order for each frame (i.e. sometimes atom 1 listed first, another time atom 10 might be listed first). If the particle IDs are included, will this affect converting the .dump file using dpdata, or does dpdata account for this? 
3.  My .dump file contains the wrapped, unscaled particle coordinates with the unscaled image flags, is this suitable for use with dpdata?
4.  The [documentation](https://docs.deepmodeling.com/projects/deepmd/en/r2/data/system.html) states to use the system energy. However, I am unsure if this means the total system energy (kinetic and potential) or just the potential energy. Can you please provide some clarification? 
5. My .dump file contains atomic positions and forces. I attempted to use dpdata to convert the data I did have in the .dump file to a format usable by DeePMD-Kit. However, I was unable to retrive the forces following the [dpdata documentation](https://docs.deepmodeling.com/projects/dpdata/en/master/systems/system.html#load-data). For reference, this is the command I used to generate my .dump file in lammps:
```
dump           my_dump all custom 1 file_name.dump id type x y z ix iy iz fx fy fz
```
Here is the script I used to try and extract the forces:
```
import dpdata
dp_output = dpdata.System("file_name.dump",fmt="lammps/dump")
d_output.to_deepmd_raw('dpmd_raw')
d_output['forces']
```
Here is the error message I receive when I run this script: _KeyError: 'forces'_ 
I was able to retrieve the atomic coordinates, and I was able to generate box.raw, type.raw, coord.raw, and type_map.raw using this script. However, the forces were not output to a .raw file.
**LAMMPS has multiple formations for generating .dump files. Does the .dump file need to be in a particular format for use with dpdata?**

I have also attached my .dump file and my file containing the system energy and virials for each frame. 
[dump_and_log_file.zip](https://github.com/user-attachments/files/17611944/dump_and_log_file.zip)

Thank you!

**Comment by njzjz** [comment]
You can parse the file independently and feed the data dictionary to dpdata. An example is

https://github.com/deepmodeling/deepmd-kit/blob/38815b371162a2153b0c2b24a38867825665c7a3/source/tests/infer/test_models.py#L291-L303

`LabeledSystem` is for `energies` and `forces`.

**Reply by agvent** [reply]
Thank you for the quick response, this is very helpful. 

However, I am still seeking some clarification regarding the energy required for each frame. Is it the total energy of the frame or the potential energy of the frame? I am using lammps to generate my training data, so should I be using the total energy of the system or the potential energy of the system?

Thank you.

**Reply by njzjz** [reply, marked as answer]
It's the potential energy that only corresponding to the atomic coordinates and types, and the forces should be the negative gradient of the potential energy.
